yt_xarray/accessor/accessor.py

```diff
@@ -15,6 +15,42 @@ def __init__(self, xarray_obj):
         self._bbox = {}
         self._field_grids = defaultdict(lambda: None)
 
+    def load_grid_from_callable(self):
+        def _read_data(handle):
+            def _reader(grid, field_name):
```

Comment:
This feels like a common usage pattern we should support somehow.  I know we did with the `load_hdf5_file` pattern ... I wonder if a slicing mechanism would be good to expose at the top level, in a way that mirrors memory mapping.

Reply:
yes! I agree. and actually, this whole function could probably be in yt proper, as a general `load_netcdf` or similar. Would still want an xarray accessor method so you could do `xr_ds.yt.to_yt()` but it could simply call the yt method.